clar2wasm/src/deserialize.rs

```diff
@@ -196,10 +202,10 @@ impl WasmGenerator {
 
                     // Allocate space for the principal on the call stack
                     let result_offset = self.module.locals.add(ValType::I32);
-                    then.global_get(self.stack_pointer).local_tee(result_offset);
+                    then.local_get(offset_result).local_tee(result_offset);
```

Comment:
We have two variables `offset_result` and `result_offset`. That seems confusing...